system/lib/wasmfs/backends/opfs_backend.cpp

```diff
@@ -297,9 +299,7 @@ class OPFSFile : public DataFile {
       default:
         WASMFS_UNREACHABLE("Unexpected open state");
     }
-    // TODO: This is the correct error code for negative read positions, but we
-    // should handle other errors correctly as well.
-    return nread < 0 ? -EINVAL : nread;
+    return nread;
```

Comment:
I guess this means we can only read/write 2GB at a time? I suppose that's ok...

Reply:
In wasm32 the `read` syscalls can only read 2GB at a time anyway because their result type is a 32-bit `ssize_t`. We should update this to use the full 63 bits in wasm64, though. I think there's a TODO for that.

Reply:
Ok, thanks. Doesn't sound urgent anyhow I guess, 31 bit reads is quite a lot...